src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -634,23 +641,24 @@ class module_elf : public module_impl
         std::string argnm{ symname, symname + std::min(strlen(symname), dynstr->get_size()) };
         auto symbol_type = static_cast<patcher::symbol_type>(rela->r_addend);
 
+        patcher::buf_type buf_type = patcher::buf_type::ctrltext;
         std::vector<uint64_t> offsets;
         offsets.push_back(ctrlcode_offset);
-        arg2patcher.emplace(std::move(argnm), patcher{ symbol_type, offsets });
+        arg2patcher.emplace(std::move(argnm), patcher{ symbol_type, offsets, buf_type});
```

Comment:
Ditto ??
```suggestion
        arg2patcher.emplace(std::move(argnm), patcher{ symbol_type, {ctrlcode_offset}, buf_type});
```

Reply:
Maybe this will work. But it has been this way for a while, And I do not see significant improvement using the suggested change either.